api.go

```diff
@@ -138,6 +138,11 @@ type Raft struct {
 	// the log/snapshot.
 	configurations configurations
 
+	// Holds a copy of the latest configuration which can be read
+	// independently from main loop.
+	latestConfiguration     Configuration
```

Comment:
Should we be returning the latest configuration or the latest _comitted_ configuration? I may well be misremembering but I feel like we shouldn't be using/returning the config until it's committed right?

Reply:
nm. `GetConfiguration` docs state:

> GetConfiguration returns the latest configuration. This may not yet be currently in use. This may not yet be committed...

SO this preserves that behaviour although frankly that seems strange to me that Consul or an application would "see" a config that tis not the one actually in use and may never get committed by a quorum 🤔.

I guess that is an edge case issue we can solve separately though this preserves same behaviour while fixing the performance issue.

Reply:
I was wondering about the same thing, but wanted to preserve current behaviour.